File: CSharp/1107.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        while (true)
        {
            int altura;
            int largura;

            int passadas = 0;
            int comparacao = 0;

            string[] entrada = Console.ReadLine().Split();
            altura = int.Parse(entrada[0]);
            largura = int.Parse(entrada[1]);

            if (altura == 0 && largura == 0)
            {
                break;
            }

            string[] medidas = Console.ReadLine().Split();
            for (int i = 0; i < largura; i++)
            {
                int xi = int.Parse(medidas[i]);

                if (i == 0)
                {
                    passadas += altura - xi;
                }
                else if (xi < comparacao)
                {
                    passadas += comparacao - xi;
                }
                comparacao = xi;
            }

            Console.WriteLine(passadas);
        }
    }
}

File: Python/1107.py
while True:
    entrada = input().split()
    altura = int(entrada[0])
    largura = int(entrada[1])

    if altura == 0 and largura == 0:
        break

    xxxx = input().split()
    passadas = 0
    comparacao = 0

    for i in range(largura):
        xi = int(xxxx[i])

        if i == 0:
            passadas += altura - xi
        elif xi < comparacao:
            passadas += comparacao - xi

        comparacao = xi

    print(passadas)
